tests/test_arrow_dataset.py

```diff
@@ -2144,6 +2144,37 @@ def test_task_question_answering(self, in_memory):
                 )
                 self.assertDictEqual(features_after_cast, dset.features)
 
+    def test_task_summarization(self, in_memory):
+        # Include a dummy extra column `dummy` to test we drop it correctly
+        features_before_cast = Features(
+            {"input_text": Value("string"), "input_summary": Value("string"), "dummy": Value("string")}
+        )
+        features_after_cast = Features({"text": Value("string"), "summary": Value("string")})
+        task = Summarization(text_column="input_text", summary_column="input_summary")
+        info = DatasetInfo(features=features_before_cast, task_templates=task)
+        data = {
+            "input_text": ["jack and jill took a taxi to attend a super duper party in the city."],
+            "input_summary": ["jack and jill attend party"],
+            "dummy": ["123456"],
+        }
+        # Test we can load from task name
+        with Dataset.from_dict(data, info=info) as dset:
```

Comment:
it seems that whatever magic @lhoestq did with windows now allows me to avoid using the annoying context manager that we have in many unit tests:

```python
with tempfile.TemporaryDirectory() as tmp_dir, Dataset.from_dict(data) as dset:
    with self._to(in_memory, tmp_dir, dset) as dset:
        # do operations on `dset`
```

i never really understood _why_ we had to do this so please correct me if it's actually needed 😄

Reply:
The `with` statement are used to properly close open arrow files on windows (it really doesn't like keeping things open).
What you mention though is the call to `self._to` call that moves a dataset on disk or in memory, which can be useful in some tests. Here you don't need to do this indeed.

Since the `in_memory` parameter is not used in this test, you can move this test outside of the `BaseDatasetTest` class and have it as a regular pytest test case.
This way it won't be run twice (once for `in_memory=True` and once for `in_memory=False`)